src/Altinn.App.Core/Helpers/PlatformHttpResponseSnapshotException.cs

```diff
@@ -0,0 +1,204 @@
+using System.Net.Http.Headers;
+using System.Text;
+
+namespace Altinn.App.Core.Helpers;
+
+/// <summary>
+/// Exception that represents a failed HTTP call to the Altinn Platform,
+/// containing an immutable snapshot of the HTTP response, while remaining
+/// backward compatible with <see cref="PlatformHttpException"/>.
+/// <para>
+/// This class derives from <see cref="PlatformHttpException"/> so existing
+/// catch blocks continue to work. It passes a sanitized, non-streaming
+/// <see cref="HttpResponseMessage"/> to the base class to avoid keeping any
+/// live network resources, and it exposes string-based snapshot properties
+/// for safe logging and persistence.
+/// </para>
+/// </summary>
+public sealed class PlatformHttpResponseSnapshotException : PlatformHttpException
```

Comment:
Could this be internal as well? Like if `PlattformHttpException` is fully usable now?